### PR TITLE
Use namespaced ansible_facts

### PR DESCRIPTION
**Pull request by cognifloyd**
If ansible's INJECT_FACTS_AS_VARS is disabled, then this role will not work. So, use the namespaced version under ansible_facts.

https://docs.ansible.com/ansible/2.8/reference_appendices/config.html#inject-facts-as-vars

### CODE REVIEW

**Comment by geerlingguy** [comment]
@cognifloyd - This is the first time I've ever seen `INJECT_FACTS_AS_VARS`; what's the use case for that? There are thousands of roles (and hundreds I maintain) using the built-in facts with `ansible_` prefix, and if this is something that's in common use, I'm surprised nobody else has ever complained about it...

Most of the blog posts and docs online refer to the `ansible_` facts too (e.g. the seminal result that is always at the top of Google results: https://raymii.org/s/tutorials/Ansible_-_Only_if_on_specific_distribution_or_distribution_version.html).

**Comment by cognifloyd** [comment]
This was introuduced in Ansible 2.5 (released in 2018). Here's the relevant section of [2.5 changelog's Major Changes](https://github.com/ansible/ansible/blob/stable-2.5/changelogs/CHANGELOG-v2.5.rst) (added in https://github.com/ansible/ansible/commit/9c629f8a1ccadd0e07c6f7ba8953a021db6a176c):

> Added fact namespacing; from now on facts will be available under `ansible_facts` namespace (for example: `ansible_facts.os_distribution`) without the `ansible_` prefix. They will continue to be added into the main namespace directly, but now with a configuration toggle to enable this. This is currently on by default, but in the future it will default to off.

Reasoning for this change is in https://github.com/ansible/ansible/pull/18445#issuecomment-259807665

> it avoids collisions with existing vars by making sure all facts are restricted to ansible_facts.

And here's a section of the [2.5 porting guide](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.5.html#ansible-fact-namespacing). However the doc change wasn't added to the 2.5 porting guide until 2.7:

- Porting guide updated in https://github.com/ansible/ansible/pull/36059. It has details on why adding to porting guide was delayed (https://github.com/ansible/ansible/commit/a892a6ef03a0b4e0e6fc7b66c20f82154cad8a8d is part of 2.8)
- examples updated to use new namesapce in https://github.com/ansible/ansible/pull/44980 (backported from devel in https://github.com/ansible/ansible/commit/ceb474bb9e95058bd0cd031e01707754377cff2c)

I follow the porting guides. As such, I switch to new behavior as soon as possible so that new things I write have the longest life span possible; I ensure that I have this setting as True when I run my playbooks so that I'm actively working with the new behavior instead of being passively surprised when the default changes on me.

This change is safe for ansible 2.5+. Only roles that need to support <2.4 (EOL) need to continue supporting the old fact names. So, the choice is to remain backwards-compatible and risk forwards-compatibility, or to switch to the namespaced facts and be fully compatible but only backwards compatible down to ansible 2.5.

If you don't want to use the namespaced vars, that's a fair choice. In that case, I will look for an alternative role, or continue to maintain another EPEL role.

**Comment by cognifloyd** [comment]
Also, the minimum ansible version was bumped in #45 from 2.4 to 2.5 to accomodate the `version` test (instead of `version_compare`). That's no longer needed since you removed the fingerprint addition, so if you still need to support 2.4, then that should be dropped back down to 2.4. If you don't need 2.4, then I hope you'll merge this; using `ansible_facts` namespace is the best-practice in all of the ansible docs.

**Comment by cognifloyd** [comment]
Rebased on master

File: defaults/main.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
 epel_repo_url: "https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ansible_distribution_major_version }}.noarch.rpm"
```
Comment (cognifloyd) [inline defaults/main.yml:2]:
```suggestion
epel_repo_url: "https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ansible_facts.distribution_major_version }}.noarch.rpm"
```

**Comment by cognifloyd** [comment]
@geerlingguy Do you have a plan for this PR? The stalebot is vociferously opposed to it.

**Pull request closed without merging**